content/en/Features/db_migration.md

```diff
@@ -0,0 +1,257 @@
+and the new data is used instead.
+3. Once the database and servers are up-to-date, you remove the old, now unused, data from the database.
+
+The 3-point migration is admittedly slower, but it cuts off the vast majority of venues for error.
+
+### Keep an Eye on Database Locks
+
+Depending on the migration tool you are using and the size of the dataset, the migration execution itself could be using 
+long-lasting locks based on the amount of data being created/updated/destroyed. This can create serious problems as the
+locks may forbid any parallel write operations, meaning your application will fail in any endpoint executing such an
+action, and cause downtime. These dangers are more difficult to track as they are typically dependent on the size of the
+database, and as staging databases are typically a fraction of the size of production, such QA will not catch the danger.
+Aside from a full production replication for testing, the only reasonable deterrent would be a database-educated engineer
+approving/guarding db migration addition.
+
+### Validate on as Valid as Data as Possible Outside of Production
+
+Even in small datasets, database migrations will always face the danger of improper data transformations which cause errors.
+One of the best countermeasures for such errors is testing the migrations in non-production databases which closely
+resemble production-- the more similar the better as it will test more potential edge cases. Typically, this is done
+as part of the deployment to a "staging/qa" environment.
+
+### Have Folks on Standby
+
+Always prepare under the assumption that if something can go wrong, it will go wrong. Hence, always have someone on 
+standby monitoring the migration.
+
+## DB Migration Options with Pure Opta
+Now that we have the preparations in place, there are several ways which database migrations can be done with Opta,
+depending on your needs/situation.
+
+### On a Server's Startup
+The simplest form is having the migration command run as part of server startup. This option is arguably the easiest/best
+to begin with and possible due to the aforementioned locking nature of migration, ensuring that even if multiple servers 
+are started simultaneously your data will not be tainted (assuming you use a tool which keeps track of the migration 
+version, which is most-if-not-all popular ones). For instance, consider the following startup script for a 
+[flask](https://flask.palletsprojects.com/en/2.0.x/) server which runs the migration and then starts the server.
+
+```shell
+set -exo pipefail
+
+export PYTHONPATH=$(pwd)
+
+source $(pipenv --venv)/bin/activate
+
+cd srv
+flask db upgrade # Does the migrations
+gunicorn -c gunicorn.py wsgi # Starts the server
+```
+
+As long as the migrations are short, this option should work fine, keep your database and server changes in lock step 
+and keep your ci/cd unchanged.
```

Comment:
the potential issue with this solution is that the probe check might fail and the pod could be killed during migration. 
We can document this risk a bit more and advise the user to set the `readiness_probe_path` to allow enough time for the migration to complete.

Reply:
done